### ISSUE TITLE
Wrong autogenerated provider.config

### ISSUE THREAD

**Issue by chipironcin** [projectatomic/atomicapp#354]
I have kubernetes installed.
I have my kubernetes cluster configuration in `~/.kube/config`
I have a Nulecule and ancillary files for my application.
I have an `answers.conf` file where I specify the provider should be kubernetes and the config file should be `~/.kube/config`
`atomicapp -a answers.conf run .` works
It creates a `provider.config` file
Doing `atomicapp -a answers.conf stop .` will try to use the autogenerated provider.config file instead of the specified in `answers.conf`

That autogenerated file should be wrong as I get the following error:
```
2015-10-22 16:04:16,777 - kubernetes - INFO - Undeploying from Kubernetes
2015-10-22 16:04:16,802 - atomicapp.utils - INFO - atomicapp.status.error.message=cmd failed: /usr/local/bin/kubectl --kubeconfig=provider.config scale rc backendless --replicas=0 --namespace=default
2015-10-22 16:04:16,802 - atomicapp.cli.main - ERROR - Exception caught: Exception('error: crypto/tls: failed to parse certificate PEM data\n',)
2015-10-22 16:04:16,802 - atomicapp.cli.main - ERROR - Run the command again with -v option to get more information.
```

Running manually:
```
/usr/local/bin/kubectl --kubeconfig=provider.config scale rc backendless --replicas=0 --namespace=default
error: crypto/tls: failed to parse certificate PEM data
```

It can be solved copying my kubernetes configuration file to provider.config. Then it will not be overriden and I can stop my application using atomicapp cli.

**Comment by dustymabe** [comment]
hi @chipironcin. so unfortunately you have confused the provider config with the answers.conf file. The answers.conf file is there to supply answers to questions that may be asked interactively to the user during deployment as well as to provide a few other config variables for atomicapp itself. 

For supplying your own kubernetes config to kube then you should use the `providerconfig` variable within your answers.conf file to specify it like is shown [here](https://github.com/projectatomic/atomicapp/blob/cdk2-beta3-rc1/docs/providers.md#kubernetes).

If you need any more help or some recommendations then drop by #nulecule on freenode and chat with us.

**Comment by dustymabe** [comment]
@chipironcin can you give an absolute path to your provider config rather than the one with the `~` and see if that works.

**Comment by chipironcin** [comment]
Let me try, thanks for reading this @dustymabe

**Comment by chipironcin** [comment]
It do the same. It reads my /home/user/.kube/config file, deploy the application succesfully but creates a bad provider.conf. Then trying to stop I got the error:
```
2015-10-26 12:46:21,304 - atomicapp.utils - INFO - atomicapp.status.error.message=cmd failed: /usr/local/bin/kubectl --kubeconfig=provider.config delete -f ./.workdir/BackendlessContainer/artifacts/kubernetes/BackendlessContainer-pod.yaml --namespace=default
2015-10-26 12:46:21,304 - atomicapp.cli.main - ERROR - Exception caught: Exception('error: could not read an encoded object from ./.workdir/BackendlessContainer/artifacts/kubernetes/BackendlessContainer-pod.yaml: unable to connect to a server to handle "pods": crypto/tls: failed to parse certificate PEM data\n',)
2015-10-26 12:46:21,304 - atomicapp.cli.main - ERROR - Run the command again with -v option to get more information.
```

Generated provider.conf and /home/user/.kube/config are not the same, althought it contains user, password, account, etc
Something to do with:
  - certificate-authority-data
  - client-certificate-data
  - client-key-data

**Comment by dustymabe** [comment]
hey @chipironcin, i'm looking at this today. We recently deleted the code that creates a provider config as we don't think it's necessary anymore (kubectl doesn't complain if one doesn't exist). In the mean time if you want to stop by #nulecule on freenode we can chat about it as well.

**Comment by dustymabe** [comment]
@chipironcin can you try this with the `projectatomic/atomicapp:dev` container? this container uses the code from the new branch that we have that does not generate a provider config as we don't feel it is needed anymore. Also what platform are you running on so I can try to simulate the same setup?

**Comment by chipironcin** [comment]
I think that would have solved the problem from root :)
I was using a specific version of `projectatomic/atomicapp` (last available, I think 0.1.3) because I had problems using `projectatomic/atomicapp:dev`
I was running it on Amazon EC2 Ubuntu instance.
As this has been solved  by not creating a provider config file, I think I can close this issue :)

Thanks for you support!

**Comment by dustymabe** [comment]
Thanks @chipironcin, let us know if you hit any other bugs!
